core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushdownLimitIntoRowNumber.java

```diff
@@ -36,6 +37,7 @@
     private static final Capture<RowNumberNode> CHILD = newCapture();
     private static final Pattern<LimitNode> PATTERN = limit()
             .matching(limit -> !limit.isWithTies() && limit.getCount() != 0 && limit.getCount() <= Integer.MAX_VALUE)
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
This should be allowed and it should cause the RowNumberNode to become order sensitive

Reply:
Making this change.
Should there be a PushdownTopNIntoRowNumber rule along similar lines ?